src/pykeen/models/meta/filtered.py

```diff
@@ -0,0 +1,217 @@
+        additional_triples: Union[None, MappedTriples, List[MappedTriples]] = None,
+        apply_in_training: bool = False,
+        base: HintOrType[Model] = "rotate",
+        **kwargs,
+    ) -> None:
+        """
+        Initialize the model.
+
+        :param triples_factory:
+            the (training) triples factory; used for creating the co-occurrence counts *and* for instantiating the
+            base model.
+        :param additional_filter_triples:
+            additional triples to use for creating the co-occurrence statistics
+        :param in_training:
+            whether to apply the masking also during training
+        :param base:
+            the base model, or a hint thereof.
+        """
+        # avoid cyclic imports
+        from .. import model_resolver
+
+        # create base model
+        base = model_resolver.make(base, triples_factory=triples_factory, pos_kwargs=kwargs)
+
+        super().__init__(
+            triples_factory=triples_factory,
+            loss=base.loss,
+            predict_with_sigmoid=base.predict_with_sigmoid,
+            random_seed=base._random_seed,
+        )
+        # assign *after* nn.Module.__init__
+        self.base = base
+
+        # index
+        h, r, t = (
+            prepare_filter_triples(
+                mapped_triples=triples_factory.mapped_triples, additional_filter_triples=additional_triples or []
+            )
+            .numpy()
+            .T
+        )
+        self.head_per_relation, self.tail_per_relation = [
+            get_csr_matrix(
+                row_indices=r,
+                col_indices=col_indices,
+                shape=(triples_factory.num_relations, triples_factory.num_entities),
+            ).astype(bool)
+            for col_indices in (h, t)
+        ]
+        self.relation_per_head, self.relation_per_tail = [
+            m.transpose().tocsr() for m in (self.head_per_relation, self.tail_per_relation)
+        ]
+
+        self.apply_in_training = apply_in_training
+
+    # docstr-coverage: inherited
+    def _get_entity_len(self, *, mode: Optional[InductiveMode]) -> Optional[int]:
+        return self.base._get_entity_len(mode=mode)
+
+    # docstr-coverage: inherited
+    def _reset_parameters_(self):
+        return self.base._reset_parameters_()
+
+    # docstr-coverage: inherited
+    def collect_regularization_term(self) -> torch.FloatTensor:  # noqa: D102
+        return self.base.collect_regularization_term()
+
+    # docstr-coverage: inherited
+    def score_hrt(
+        self, hrt_batch: torch.LongTensor, *, mode: Optional[InductiveMode] = None
+    ) -> torch.FloatTensor:  # noqa: D102
+        if self.apply_in_training:
+            raise NotImplementedError
+        return self.base.score_hrt(hrt_batch=hrt_batch, mode=mode)
+
+    # docstr-coverage: inherited
+    def score_h(
+        self, rt_batch: torch.LongTensor, *, slice_size: Optional[int] = None, mode: Optional[InductiveMode] = None
```

Comment:
might be better to use **kwargs after rt_batch (same for all of the other ones that delegate)

Reply:
https://github.com/pykeen/pykeen/pull/943/commits/9719a64c62583d028c7199d5cc39900fd9912de2